### PR TITLE
JIT: Support block statements

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: jit-compiler/src/codegen.rs [hunk 1]
```diff
@@ -312,9 +312,27 @@ impl<'a, T: FieldElement> CodeGenerator<'a, T> {
...
+                };
+                let value = self.format_expr(value)?;
+                let var_name = "scrutinee__";
+                let ty = ty
+                    .as_ref()
+                    .map(|ty| format!(": {}", map_type(ty)))
+                    .unwrap_or_default();
+
+                let (vars, code) = check_pattern(var_name, pattern)?;
+                // TODO if we want to explicitly specify the type, we need to exchange the non-captured
```
Comment (chriseth) [inline jit-compiler/src/codegen.rs:330]: This is just a note in case we need it in the future if rust cannot determine the type.

Comment (lvella) [reply jit-compiler/src/codegen.rs:330]: I don't understand this comment. It is talking about the type of `{vars}` , I assume. But what are the non-captured parts that must be exchanged by `()`?

**Review by lvella — APPROVED** [review]
Approved, unless you agree the TODO is a little cryptic.

**Comment by chriseth** [comment]
I'll create a follow-up PR to extend the todo.

**Pull request merged**